integrated_channels/cornerstone/views.py

```diff
@@ -12,6 +12,112 @@
 from integrated_channels.cornerstone.models import CornerstoneEnterpriseCustomerConfiguration
 
 
+class BaseViewSet(generics.ListAPIView):
+    """
+    Base class for all Cornerstone view sets.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+
+    OAuth2Authentication = get_oauth2authentication_class()
+    if OAuth2Authentication is not None:
+        authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    else:
+        authentication_classes = (JwtAuthentication, SessionAuthentication,)
+
+    throttle_classes = (ServiceUserThrottle,)
+    renderer_classes = [renderers.JSONRenderer]
+
+
+class CornerstoneCoursesUpdates(BaseViewSet):
+    """
+        **Use Cases**
+
+            Get a list of courses that need updated to share on cornerstone filtered by an enterprise customer.
+
+        **Example Requests**
+
+            GET /cornerstone/course-updates?ciid={customer_uuid}
```

Comment:
are we going to tell Cornerstone customers to use this new endpoint? Or did I miss it being used elsewhere in this PR? This is as dumb of a question as it looks :D

Reply:
No dumb questions 🙃. The idea is that we can't change the behaviors of existing endpoints because customers are expecting and handling the current patterns. So I added this new endpoint that we can roll out to customer's when they change their systems to handle it. There's a monthly meeting between CSOD and Matt D and the team- I'm going to be joining that call to talk about the new endpoint and any potential improvements that could be made (transmitting formatted data like `updated courses`, `created courses` and `deleted courses`. But until that conversation happens, this will basically be unused.